MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -356,6 +395,126 @@ namespace WPEFramework {
+                        if (joGetResult.HasLabel("success") && joGetResult["success"].Boolean()) {
+                            if (joGetResult.HasLabel("partnerProvisioningContext")) {
+                                JsonObject getProvisioningContext = joGetResult["partnerProvisioningContext"].Object();
+                                int size = (int)(sizeof(deviceInitializationContext)/sizeof(deviceInitializationContext[0]));
+                                for (int i=0; i < size; i++) {
+                                    const char* key = deviceInitializationContext[i].c_str();
+
+                                    // Retrive partnerProvisioningContext Value
+                                    string paramValue = getProvisioningContext[key].String();
+
+                                    if(!paramValue.empty()) {
+                                        if (strcmp(key, "regionalConfigService") == 0) {
+                                            paramValue = "https://" + paramValue;
+                                        }
+                                        LOGINFO("%s : %s", key, paramValue.c_str());
+
+                                        // Retrieve tr181 parameter from m_param_map
+                                        string rfc_parameter = m_param_map[key];
+
+                                        //  Retrieve parameter data type from m_paramType_map
+                                        DATA_TYPE rfc_dataType = m_paramType_map[key];
+
+                                        // Set the RFC values for partnerProvisioningContext parameters
+                                        setRFC(rfc_parameter.c_str(), paramValue.c_str(), rfc_dataType);
+                                    } else {
+                                        LOGINFO("Not able to fetch %s value from partnerProvisioningContext", key);
+                                    }
+                                }
+                                success = true;
+                            } else {
+                                LOGINFO("partnerProvisioningContext is not available in the response. Retrying in %d seconds", retryDelay);
+                                sleep(retryDelay);
+                            }
+                        } else {
+                            // Get retryDelay value and sleep for that much seconds
+                            if (joGetResult.HasLabel("retryDelay")) {
+                                retryDelay = joGetResult["retryDelay"].Number();
+                            }
+                            LOGINFO("getDeviceInitializationContext failed. Retrying in %d seconds", retryDelay);
+                            sleep(retryDelay);
+                        }
+                    }
+                } else {
+                    LOGINFO("%s is not active, Retrying in 5 seconds", secMgr_callsign);
+                    sleep(5);
+                }
+            } while (!success);
+            return success;
```

Comment:
This looks like indefinite loop. Checking why do we need it? Can you pls check and cap the number of retries? Why doesn't getDeviceInitializationContext return the correct context the first time? Is it dependent on something else in SecManager getting initialized?

Reply:
Can we have this either as a dependency at plugin level or as an event based trigger for Unsolicited maintenance?.